c++/src/kj/string.h

```diff
@@ -276,6 +276,8 @@ class String {
   inline constexpr size_t size() const;
   // Result does not include NUL terminator.
 
+  inline constexpr void shrink(size_t size);
```

Comment:
Some code comments here explaining the semantics here would be good. e.g. does this cause a reallocation, etc